contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -3637,13 +3655,17 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
 								foreach(lc, grant->grantees)
 								{
 									RoleSpec	   *rol_spec = (RoleSpec *) lfirst(lc);
-									int i = 0;
-									char *permissions[] = {"select", "insert", "update", "references", "delete"};
-									for(i = 0; i < 5; i++)
+									if((rol_spec->rolename != NULL) && (strcmp(rol_spec->rolename, "public") != 0) && !check_bbf_schema_for_entry(logical_schema, obj,rol_spec->rolename))
+									{
+										add_entry_to_bbf_schema(logical_schema, obj, 31, rol_spec->rolename, obj_type);
+									}
+									else if(rol_spec->rolename != NULL && (strcmp(rol_spec->rolename, "public") != 0))
```

Comment:
Instead of writing
```
if (A && B && C)
{
}
else if (A && B)
{
}
```
please reformat the same as
```
if (A && B)
{
   if (C)
   {
   }
  else
   {
   }
}
```